posts/2022-01-14-construct-vendia-univseral-application-using-domain-driven-design.md

```diff
@@ -0,0 +1,177 @@
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029339-3c97b4d4-e66a-4b37-ae21-6200e22cb042.png" />
+</p>
+<p align="center">Figure 3 - Uni Development Lifecycle Stages</p>
+
+
+## Uni Schema Anatomy
+
+Let’s imagine a simple Uni schema that provides a datastore for core customer data attributes. Figure 4 breaks the Uni schema into parts and tries to map it to concepts you use in a relational database. This will help you quickly get started on building your own Uni. 
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029529-fa688c30-766c-4803-bc59-3b953e3ae0c4.png" />
+</p>
+<p align="center">Figure 4 - Example Customer Uni</p>
+
+
+## Applying DDD to Uni 
+
+
+### Pattern 1: Single Domain, Single Bounded Context, and a Single Uni
+
+This pattern is well suited for organizations who want to build small custom applications surrounding complex enterprise applications, or organizations who want to construct a new application to prove a new business idea. 
+
+In this pattern, a Uni model contains a data model for a single bounded context. For example, the CRM team models customer account details in Bounded Context 1. 
+
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029629-68861fc8-daaf-4dc5-ab21-5269df0a5068.png" />
+</p>
+<p align="center">Figure 5 - Pattern 1</p>
+
+**You can use the below resources to practice and deploy pattern 1.**
+
+<details>
+<summary>Schema</summary>
+[schema.json](https://github.com/vendia/examples/blob/7792069dd40c6822279305b201965685aaac5a08/share/data-modeling/domain-driven-design/pattern1/schema/schema.json)
+</details>
+<details>
+<summary>Details and deployment instructions</summary>
+[README.md](https://github.com/vendia/examples/blob/vikrant-data-modeling-blog/share/data-modeling/domain-driven-design/pattern1/README.md)
+</details>
+
+
+### Pattern 2: Single Domain, Multiple Bounded Contexts, and a Single Uni
+
+This pattern is well suited for organizations who already have experience working with Uni and want to leverage existing Uni implementation in their organization with more custom and new applications. This model is also a good fit for new analytics focussed domains within the organization that has not built its software platform yet. These organizations can continue to operate lean by leveraging Uni owned by other domains with read-only copies of data for analytics and visualization. 
```

Comment:
A great use case here (and one we're having active customer conversations around) is building modern web/mobile applications that span many legacy systems.  Using Share as an aggregation point (through single or bi-directional adapters).  It makes multiple domains accessible under a single Graphql API, leading to much faster application development as a result.

Reply:
great point.